src/irmin-pack/unix/lower.ml

```diff
@@ -148,14 +148,19 @@ module Make_volume (Io : Io.S) (Errs : Io_errors.S with module Io = Io) = struct
         | None -> Errs.raise_error (`Invalid_volume_read (`Closed, off))
         | Some s -> Sparse.read_range_exn s ~off ~min_len ~max_len b)
 
-  let archive_seq ~upper_root ~generation ~is_first ~to_archive ~off t =
+  let archive_seq ~upper_root ~generation ~is_first ~to_archive t =
     let open Result_syntax in
     let root = path t in
+    let* first_off =
+      match to_archive with
+      | [] -> Error `Volume_nothing_to_archive
```

Comment:
Will this trigger in the case after a migration and the GC commit is in the lower? I don't think we'd want that GC to fail.

Reply:
I remember us talking about it working that way, but I haven't double-check the upstream logic to make sure we do indeed short-circuit the process. Either way, I think I slightly prefer the lower not returning an error for this case so a warning is good.